### PR TITLE
add babel support

### PR DESCRIPTION
**Pull request by genofire**

### CODE REVIEW

**Comment by rubo77** [comment]
There are lots of changes, please rebase this branch to the latest master

**Comment by genofire** [comment]
I will, but I need to support babel and batman side by side.
So we need another way for generating `node_id`.

I also hope that a multi side batman  would be merged before ... during iteration of a list of with zero batman interfaces would solve many parts.

File: providers/neighbours/babel.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    		               "protocol": "babel",
+                           "neighbours": ( lambda result, ifname: {
+                            data[4]: {
+                                "rxcost": int(data[12]),
+                                "txcost": int(data[14]),
+                                "cost": int(data[16]),
+                                "reachability": int(data[8],16)
+                            }
+                            for data in result
+                            if data[0] == "add" and data[1] == "neighbour" and data[3] == "address" and data[5] == "if" and data[6] == ifname and data[7] == "reach" and data[11] == "rxcost" and data[13] == "txcost" and data[15] == "cost"
```
Comment (genofire) [inline providers/neighbours/babel.py]: Maybe more beautiful:
```python3
    [
(lambda y: {
  y[x]:y[x+1]
  for x in range(0,len(y),2)})
})(x.split())
for x in input.split('\n') if x != 'ok' and x != ''
]
```

**Comment by genofire** [comment]
@rubo77 Would you review and merge?

**Comment by rubo77** [comment]
I cannot rewiew this, I don't have the expertise, can please someone review this?

**Comment by herbetom** [comment]
Is this PR still of interest? A lot has changed since it was created.

**Comment by AiyionPrime** [comment]
This still looks pretty dead.

**Comment by herbetom** [comment]
yeah, probably should be closed

**Comment by genofire** [comment]
sorry - currently i have no time and motivation to fix it pull request.

**Comment by AiyionPrime** [comment]
Offer will still stand, when you come back to this:
Setup a node running gluon master with working babel and we'll use it as template to integrate babel support in mesh-announce, like we did with with WireGuard before.

**Pull request closed without merging**